Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -268,24 +250,26 @@ public override void Disable()
         {
             base.Disable();
 
-            // Input System adds a gaze provider component on the main camera, which needs to be removed when the input system is disabled/removed.
-            // Otherwise the component would keep references to dead objects.
-            // Unity's way to remove component is to destroy it.
-            if (GazeProvider != null)
+            // Shutdown of the focus provider is managed by the MixedRealityToolkit
+            // object - it's possible that the service can have been unregistered
+            // by the point that input is getting disabled and then destroyed.
+            if (Registrar.IsServiceRegistered<IMixedRealityGazeProvider>())
```

Comment:
Wait, if MRTK is managing the focus provider, why are you disabling it here? Is it because when input system is disabled, we actually want to also disable focus provider?

Ah, reading your comments it sounds like yes. So currently disabling input system...

- Disable Focus Provider
- Disable Gaze Provider

I can see why we want to disable focus provider if we disable input system (no longer want to send focus enter / exit events), but why the gaze provider? If we are disabling the gaze provider, we might as well disable every single data provider (hands, eyes, controllers). Currently our input system is actually doing not only input dispatch (dispatch focus, dispatch input events), but also dispatchign sensor data (gaze, hands, controllers). I actually think what we want eventually is to decouple providing sensor data (eyes, hands, controlelrs) from input dispatch. A first step is to only disable input stuff when input system is disabled. So I would only disable the focus provider, not the gaze provider (which is sensor data)

Reply:
Chatted offline on this one, and I think this comment comes down to different definitions and expectations for what "disable/enable" means when applied to the a larger system (in this case, the input system).

The expectation that I come from is that calling NameOfSystem.Disable() will disable everything under that system (including both dispatching and sensor data, because dispatch and sensor data are part of the input system).

While I think that this approach is consistent with the way input is written today (i.e. everything under input disabled/enabled as a unit), @julenka raised a good point on making sure that we double check consumer scenarios to make sure that we're doing the right thing - even if this ends up being more "correct" according to current designs it doesn't matter if it doesn't help consumers. I'm gonna ask around here on the various issues this is intended to address.